### PR TITLE
Revised Read Generic Network Signal data

### PR DESCRIPTION
**Pull request by atiwari9**
This proposal is to make vehicle data APIs (GetVehicleData/SubscribeVehicleData/UnsubscribeVehicleData/OnVehicleData) more flexible to allow additional new vehicle data items, while maintaining the security access via policies.

### CODE REVIEW

**Comment by atiwari9** [comment]
@jordynmackool  - I have updated the proposal with revisions as per this [comment](https://github.com/smartdevicelink/sdl_evolution/issues/502#issuecomment-465775412). Please review and merge

File: proposals/0173-Read-Generic-Network-Signal-data.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ The proposed Policy Table structure assumes that SDL enums are defined in the RP
...
 
-* OEM's schema file should always be baselined with latest version of standardized schema file from open SDL. OEM can then further add OEM specific custom vehicle data items.
+* OEM's schema file should always be baselined with latest version of standardized schema file from open SDL. OEM can then further add OEM specific custom vehicle data items. This ensures that any version of core which implement this proposal would be able to process the latest version of standardized vehicle data items schema, hence would be able to support all future additions for new standardized vehicle data items.
```
Comment (jordynmackool) [inline proposals/0173-Read-Generic-Network-Signal-data.md]: Please update: 

This ensures that any version of core which **implements** this proposal would be able to process the latest version of standardized vehicle data **item** schema, **and** hence would be able to support all future additions for new standardized vehicle data items.

Comment (atiwari9) [reply proposals/0173-Read-Generic-Network-Signal-data.md]: Done

**Comment by atiwari9** [comment]
@jordynmackool  - Updated the PR as per review comments

File: proposals/0173-Read-Generic-Network-Signal-data.md [hunk 2]
```diff
@@ -35,12 +45,23 @@ Instead of validating vehicle data items against APIs XML, SDL would rely on a d
...
 
 #### Schema file location and updates
 A base JSON schema file for SDL core and proxy will be kept locally and an additional one in the OEM cloud. OEMs can update the schema file with OEM specific items, which SDL core would be able to download over the cloud along with policy updates; SDL proxy would still process the local(static) copy of schema for existing data items. For OEM specific additional items, the OEM app would process a generic _Object_ to retrieve vehicle data item response _struct_. This new design would ensure flexibility to add OEM specific vehicle data items while maintaining access control via policies the same way it is done currently.
 
+* OEM's schema file should always be baselined with latest version of standardized schema file from open SDL. OEM can then further add OEM specific custom vehicle data items. This ensures that any version of core which implements this proposal would be able to process the latest version of standardized vehicle data item schema, and hence would be able to support all future additions for new standardized vehicle data items.
```
Comment (jordynmackool) [inline proposals/0173-Read-Generic-Network-Signal-data.md]: Update to:

OEM's schema file should always be baselined with latest version of standardized schema file from open SDL. **OEMs** can then further add OEM specific custom vehicle data items. This ensures that any version of core **that** implements this proposal would be able to process the latest version of standardized vehicle data item schema, and hence would be able to support all future additions for new standardized vehicle data items.

File: proposals/0173-Read-Generic-Network-Signal-data.md [hunk 3]
```diff
@@ -329,15 +414,34 @@ public Object getGenericNetworkData(String vehicleDataName){
...
+
 ## Impact on existing code
 * SDL core would need to add support to download and parse the new JSON schema for vehicle data. SDL would either keep the existing vehicle data items in HMI/Mobile API and only use this approach for new vehicle data items or totally replace the way vehicle data items are compiled and validated by SDL. Instead of using the hard coded xml file, valid vehicle data type would be fetched from mapping file over the cloud. (there would be a local copy for the initial build). The interface between SDL and HMI needs to be updated while passing _GetVehicleData/SubscribeVehicleData/UnsubscribeVehicleData/OnVehicleData_ requests to include return data type and reference keys.
 
 * APIs would need a new Request/Response API to act as the Generic Request Response for _GetVehicleData/SubscribeVehicleData/UnsubscribeVehicleData_
 
+* SDL_Server project needs updates as defined above
```
Comment (jordynmackool) [inline proposals/0173-Read-Generic-Network-Signal-data.md]: Please add a period

Comment (atiwari9) [reply proposals/0173-Read-Generic-Network-Signal-data.md]: Done

File: proposals/0173-Read-Generic-Network-Signal-data.md (hunk 2 above)
Comment (atiwari9) [reply proposals/0173-Read-Generic-Network-Signal-data.md]: Done

File: proposals/0173-Read-Generic-Network-Signal-data.md [hunk 4]
```diff
@@ -35,12 +45,23 @@ Instead of validating vehicle data items against APIs XML, SDL would rely on a d
...
 
+* OEM's schema file should always be baselined with latest version of standardized schema file from open SDL. OEM can then further add OEM specific custom vehicle data items.
+* VehicleData item availability and definitions in the official SDL RPC Spec will always take precedence over custom ("proprietary") parameters.
+* VehicleData items in the official SDL RPC Spec are immutable and therefore cannot be extended with proprietary data
+* All VehicleData items must be defined in the Policy Table (not just the custom/proprietary items), per recommendation by PM's Core team
```
[the remark this answers is not in the document]
Comment (atiwari9) [reply proposals/0173-Read-Generic-Network-Signal-data.md]: Done

**Comment by atiwari9** [comment]
@jordynmackool - Updated the PR as per review comments

File: proposals/0173-Read-Generic-Network-Signal-data.md [hunk 5]
```diff
@@ -51,82 +72,106 @@ A base JSON schema file for SDL core and proxy will be kept locally and an addit
...
-			"id":"messageName",
-			"dataType":"String",
-			"reference":"OEM_REF_MSG",
-			"array":false,
-			"mandatory":true
+			"name":"messageName",
+			"type":"String",
+			"key":"OEM_REF_MSG",
+			"array":true,
+			"mandatory":true,
```
[the remark this answers is not in the document]
Comment (atiwari9) [reply proposals/0173-Read-Generic-Network-Signal-data.md]: @theresalech - Custom sub param had been added to revision commit # [3b7ea1f](https://github.com/smartdevicelink/sdl_evolution/pull/684/commits/3b7ea1ff9a837e78641b58e88b2ee74f93eb8e87) . Check following section:
> ### Update VehicleDataResult to support custom vehicle data items by adding:

File: proposals/0173-Read-Generic-Network-Signal-data.md [hunk 6]
```diff
@@ -273,11 +337,14 @@ Sample response from module:
...
-	"id":"messageName",
-	"dataType":"String",
-	"reference":"OEM_REF_MSG",
-	"array":false,
-	"mandatory":true
+	"name":"messageName",
+	"type":"String",
+	"key":"OEM_REF_MSG",
+	"array":true,
+	"mandatory":true,
```
[the remark this answers is not in the document]
Comment (atiwari9) [reply proposals/0173-Read-Generic-Network-Signal-data.md]: Updated in commit [d44cc09](https://github.com/smartdevicelink/sdl_evolution/pull/684/commits/d44cc096ac39169e85e34fc5056042bd555a0b33)

**Comment by atiwari9** [comment]
@theresalech - Ready for review

**Pull request merged**